### PR TITLE
fix(ingest/kafka):add poll for admin client for oauth_cb

### PR DESCRIPTION
**Pull request by mayurinehate**
Followup on https://github.com/datahub-project/datahub/pull/11900 

## Checklist

- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [ ] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

### CODE REVIEW

File: metadata-ingestion/src/datahub/ingestion/source/kafka/kafka.py [hunk 1]
```diff
@@ -164,6 +164,25 @@ def get_kafka_consumer(
     return consumer
 
 
+def get_kakfa_admin_client(
```
Comment (treff7es) [inline metadata-ingestion/src/datahub/ingestion/source/kafka/kafka.py:167]:
```suggestion
def get_kafka_admin_client(
```

File: metadata-ingestion/src/datahub/ingestion/source/kafka/kafka.py [hunk 2]
```diff
@@ -278,13 +297,7 @@ def __init__(self, config: KafkaSourceConfig, ctx: PipelineContext):
...
         try:
             # TODO: Do we require separate config than existing consumer_config ?
-            self.admin_client = AdminClient(
-                {
-                    "group.id": "test",
-                    "bootstrap.servers": self.source_config.connection.bootstrap,
-                    **self.source_config.connection.consumer_config,
-                }
-            )
+            self.admin_client = get_kakfa_admin_client(self.source_config.connection)
```
Comment (treff7es) [inline metadata-ingestion/src/datahub/ingestion/source/kafka/kafka.py:300]:
```suggestion
            self.admin_client = get_kafka_admin_client(self.source_config.connection)
```

File: metadata-ingestion/src/datahub/ingestion/source/kafka/kafka.py [hunk 3]
```diff
@@ -164,6 +164,25 @@ def get_kafka_consumer(
     return consumer
 
 
+def get_kakfa_admin_client(
+    connection: KafkaConsumerConnectionConfig,
+) -> AdminClient:
+    client = AdminClient(
+        {
+            "group.id": "test",
```
Comment (treff7es) [inline metadata-ingestion/src/datahub/ingestion/source/kafka/kafka.py:172]: Shouldn't this be some random group id?
`test` seems to be too generic, what if it already exists?

**Review by treff7es — APPROVED** [review]

File: metadata-ingestion/src/datahub/ingestion/source/kafka/kafka.py (hunk 3 above)
Comment (mayurinehate) [reply metadata-ingestion/src/datahub/ingestion/source/kafka/kafka.py:172]: In this case, we actually do not consume any messages so any group id is fine, I think.

Comment (treff7es) [reply metadata-ingestion/src/datahub/ingestion/source/kafka/kafka.py:172]: I think this should work as it is datahub-specific, so there is less chance to collide

**Pull request merged**